### PR TITLE
CAMS-383: Case Search

### PR DESCRIPTION
**Pull request by jamesobrooks**
# Purpose

Provide the ability to search for cases across districts by case number.

# Major Changes

Adds search screen.

# Testing/Validation

- [x] Added unit tests
- [x] Added a11y test
- [x] Validated manually

### CODE REVIEW

File: user-interface/src/data-verification/ConsolidationOrderModal.tsx [hunk 1]
```diff
@@ -235,9 +235,8 @@ function ConsolidationOrderModalComponent(
     setConsolidationType(ev.target.value as ConsolidationType);
   }
 
-  async function handleLeadCaseInputChange(ev: React.ChangeEvent<HTMLInputElement>) {
-    const { caseNumber, joinedInput } = validateCaseNumberInput(ev);
-    leadCaseNumberRef.current?.setValue(joinedInput);
+  async function handleLeadCaseInputChange(caseNumber?: string) {
+    //leadCaseNumberRef.current?.setValue(joinedInput);
```
Comment (btposey) [inline user-interface/src/data-verification/ConsolidationOrderModal.tsx:239]: If not needed, delete.

Comment (jamesobrooks) [inline user-interface/src/data-verification/ConsolidationOrderModal.tsx:239]:
```suggestion
```

File: user-interface/src/lib/components/CaseNumberInput.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
+import React, { forwardRef, useImperativeHandle, useRef, useState } from 'react';
+import Input, { InputProps } from './uswds/Input';
+import { InputRef } from '../type-declarations/input-fields';
+
+const BLANK = '';
```
Comment (btposey) [inline user-interface/src/lib/components/CaseNumberInput.tsx:5]: We use '' and BLANK in the source. We should pick one for consistency. Probably just drop out BLANK.

File: user-interface/src/lib/components/Header.test.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+import * as FeatureFlags from '@/lib/hooks/UseFeatureFlags';
+
+describe('Header', () => {
+  vi.spyOn(FeatureFlags, 'default').mockReturnValue({
+    'transfer-orders-enabled': true,
+    'case-search-enabled': true,
+  });
+
+  test('should be rendered', async () => {
+    render(
```
Comment (btposey) [inline user-interface/src/lib/components/Header.test.tsx:14]: Maybe wrap this in a renderWithoutProps function???

File: user-interface/src/lib/components/cams/GenericTable/GenericTable.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  TableHeaderData,
+  TableProps,
+  TableRow,
+  TableRowData,
+} from '../../uswds/Table';
+
+const defaultTransformer = (v: unknown) => v?.toString() ?? '';
+
+/**
+ * name: Column name. Used to coorelate transformers applied to a given column.
```
Comment (btposey) [inline user-interface/src/lib/components/cams/GenericTable/GenericTable.tsx:14]: Comments lie. This isn't the case anymore.

File: user-interface/src/lib/components/uswds/Table.test.tsx
Comment (jamesobrooks) [inline user-interface/src/lib/components/uswds/Table.test.tsx:1]: Let's revisit this file to make sure we're happy with organization and the state of the tests (e.g. at least one section has no expects).

File: user-interface/src/lib/components/uswds/Table.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+//    usa-table--sticky-header
+//    usa-table--striped
+export type UswdsTableStyle =
+  | 'scrollable'
+  | 'borderless'
+  | 'compact'
+  | 'stacked'
+  | 'stacked-header'
+  | 'sticky-header'
+  | 'striped';
```
Comment (jamesobrooks) [inline user-interface/src/lib/components/uswds/Table.tsx:109]: I'd prefer to only add the ones that we are using.

File: user-interface/src/lib/components/uswds/Table.tsx
Comment (jamesobrooks) [inline user-interface/src/lib/components/uswds/Table.tsx:1]: Sort is not working.

File: user-interface/src/lib/hooks/UseApi.ts [hunk 6]
```diff
@@ -45,15 +45,22 @@ export interface ApiClient {
 
 export interface GenericApiClient {
   get<T = object>(path: string, options?: ObjectKeyVal): Promise<T>;
+  post<T = object>(path: string, body: object, options?: ObjectKeyVal): Promise<T>;
 }
 
 //This allows us to use generics and avoid typing using the "as" keyword to specify return types throughout the rest of the application
+// TODO: Need to better handle the reponse
```
Comment (jamesobrooks) [inline user-interface/src/lib/hooks/UseApi.ts:52]:
```suggestion
// TODO: Need to better handle non-200 responses.
// e.g. a 202 or 204 would result in an error.
// We are also swallowing any meaningful error responses.
```

File: user-interface/src/lib/models/chapter15-mock.api.cases.ts [hunk 7]
```diff
@@ -247,4 +247,29 @@ export default class Chapter15MockApi extends Api {
     };
     return Promise.resolve(response);
   }
+
+  public static async post(path: string, data: object, _options?: ObjectKeyVal) {
```
Comment (jamesobrooks) [inline user-interface/src/lib/models/chapter15-mock.api.cases.ts:251]:
```suggestion
  // TODO: add handling of other uses of POST (e.g. case assignment creation)
  public static async post(path: string, data: object, _options?: ObjectKeyVal) {
```

**Review by btposey — APPROVED** [review]

**Pull request merged**